src/profile_manager/views.py

```diff
@@ -149,6 +156,71 @@ def dispatch(self, *args, **kwargs):
         else:
             raise Http404("Sorry, this profile isn't yours!")
 
+    def post(self, request, *args, **kwargs):
+        forms = self.get_forms()
+
+        # check if all form instances are valid else ...
+        if all([form.is_valid() for form in forms]):
+            return self.form_valid(forms)
+        return self.form_invalid(forms)
+
+    def get_context_data(self, **kwargs):
+        profile = self.get_object()
+        context = {}
+
+        # return instance of form or new form instance
+        context['forms'] = kwargs.get('forms', self.get_forms())
+
+        context['heading'] = "Editing " + profile.user.get_username() + "'s Profile"
+        context['submit_btn_value'] = "Update"
+        context['profile'] = profile
+
+        return context
+
+    # handles form_kwargs for all forms  # redundant?
+    def get_form_kwargs(self, initial={}):
+        kwargs = initial
```

Comment:
Is there a reason you aren't calling super in this override method? `kwargs = super().get_form_kwargs()`  That initial stuff should be unnecessary.

Reply:
initially overrode ```get_form_kwargs``` since i didnt know what the initial args would do to the form. Initial is necessary because ```instance``` kwarg gave the wrong instance.

Changed

Reply:
im now just calling super().get_form_kwargs() and updating the instance in both get_form functions